enterprise/server/auth/auth.go

```diff
@@ -224,11 +224,13 @@ type authenticator interface {
 }
 
 type oidcAuthenticator struct {
-	oauth2Config *oauth2.Config
-	oidcConfig   *oidc.Config
-	provider     *oidc.Provider
-	issuer       string
-	slug         string
+	oauth2Config  func() (*oauth2.Config, error)
+	_oauth2Config *oauth2.Config
```

Comment:
uber nit: might be slightly clearer to replace `_` with `cached`

Reply:
I like this.